src/main/java/com/github/splendor_mobile_game/game/model/User.java

```diff
@@ -11,12 +17,104 @@ public class User {
 
     private int connectionHasCode;
 
+    private int points;
+
+    //initialized tokens hashmap
+    private Map<TokenType, Integer> tokens = new HashMap<TokenType, Integer>();
+
+    //initialized purchased and reserved cards lists
+    private ArrayList<Card> purchasedCards = new ArrayList<Card>();
+    private ArrayList<Card> reservedCards = new ArrayList<Card>();
+
+    //initialized nobles list
+    private ArrayList<Noble> visitingNobles = new ArrayList<Noble>();
+
     public User(UUID uuid, String name, int connectionHasCode) {
         this.uuid = uuid;
         this.name = name;
         this.connectionHasCode = connectionHasCode;
+
+        //putting every token type into the hashmap and setting its value to 0
+        for (TokenType type : TokenType.values()) {
+            this.tokens.put(type, 0);
+        }
+    }
+
+    //method returning how many tokens user has
+    public int getTokenCount() {
+        int result = 0;
+
+        for(Map.Entry<TokenType, Integer> set : this.tokens.entrySet()) {
+            result += set.getValue();
+        }
+
+        return result;
+    }
+
+    public int getTokenCount(TokenType type) {
+        return this.tokens.get(type);
+    }
+
+    //method for adding two tokens (still needs to be validated if there are at least 4 tokens of this type on the table)
+    public void takeTwoTokens(TokenType type) throws Exception {
+        if(this.getTokenCount() > 8) throw new Exception("You can't have more than 10 tokens");
+        this.tokens.put(type, this.tokens.get(type) + 2);
+    }
+
+    //method for taking tokens from user
+    public void putDownTokens(TokenType type, int amount) throws Exception {
+        if(this.tokens.get(type) - amount < 0) throw new Exception("You can't have less than 0 tokens");
+        this.tokens.put(type, this.tokens.get(type) - amount);
     }
 
+    //method for adding three different tokens
+    public void takeThreeTokens(TokenType type1, TokenType type2, TokenType type3) throws Exception {
+        if(type1 == type2 || type1 == type3 || type2 == type3) throw new Exception("No three different token types selected");
+        if(this.getTokenCount() > 7) throw new Exception("You can't have more than 10 tokens");
+        this.tokens.put(type1, this.tokens.get(type1) + 1);
+        this.tokens.put(type2, this.tokens.get(type2) + 1);
+        this.tokens.put(type3, this.tokens.get(type3) + 1);
+    }
+
+    //method for buying cards
+    public void buyCard(Card card) throws Exception {
+
+        for(Map.Entry<TokenType, Integer> set : this.tokens.entrySet()) {
+            if(set.getKey() == TokenType.GOLD_JOKER) continue;
+            if(set.getValue() < card.getCost(set.getKey())) throw new Exception("You don't have enough tokens to buy this card");
```

Comment:
There is a possibility that we may not have enough tokens of a particular type to buy a card. However, we can use gold tokens to make up for the shortage.

For instance:

1. Let's say we want to purchase a card that requires 2 red tokens and 3 green tokens.
2. Our player currently has 2 red tokens, 2 green tokens, and 1 gold token.
3. To buy this card, the player can use 2 red tokens, 2 green tokens, and 1 gold token as payment.

Because, gold tokens can be used in place of every other.

Reply:
That's a fact. Also we need to consider that golden tokens might be used only in the next turn. (I don't know if we are gonna check it here or inside a reaction)

Reply:
The next turn you mean.... 

1. We reserve card getting ne gold token, and the same time we end our turn, because we cannot do any other action
2. We wait for other players to make their turns
3. This is our next turn and we can use our gold token